lib/Transforms/Scalar/DxilConditionalMem2Reg.cpp

```diff
@@ -110,6 +110,134 @@ class DxilConditionalMem2Reg : public FunctionPass {
     AU.setPreservesCFG();
   }
 
+  // Replace simple array allocas with individual scalar allocas.
+  // Only handle if:
+  // - All the alloca's users are geps
+  // - The geps all have only constant indices
```

Comment:
Ah! This is why you changed the existing tests to use non-constant indices right?

Reply:
Yep.